scripts/Tools/xmlchange

```diff
@@ -143,8 +143,19 @@ def xmlchange_single_value(case, xmlid, xmlval, subgroup, append, force, dryrun)
         xmlval = convert_to_type(xmlval, type_str, xmlid)
 
     if not dryrun:
-        newval = case.set_value(xmlid, xmlval, subgroup, ignore_type=force)
-        expect(newval is not None,"No variable \"%s\" found"%xmlid)
+        result = case.set_value(xmlid, xmlval, subgroup, ignore_type=force, return_file=True)
+        expect(result is not None,"No variable \"%s\" found"%xmlid)
+        filename = result[1]
+
+        if filename.endswith("env_batch.xml"):
+            logger.info("case.submit will auto-regen your batch files, no further user action is needed")
+        elif filename.endswith("env_build.xml"):
+            logger.info("You should consider cleaning and rebuilding your case")
+        elif filename.endswith("env_mach_pes.xml"):
+            logger.info("Changing mach_pes can have a big impact on your case. Please re-run case.setup, and clean/rebuild your case")
+        elif filename.endswith("env_case.xml"):
```

Comment:
I thought you're not allowed to change something in env_case, ever. Is it really safe to change these values if you rerun case.setup and rebuild?

Reply:
I was wondering that myself. @jedwards4b ?

Reply:
Right - you should never change anything in env_case.xml -  I think there is a way to mark that file as read only.    There are a few special circumstances where you may want to change something in this directory but I'm not sure it'll work - say for example you need to move a case directory.